staging/src/k8s.io/legacy-cloud-providers/gce/gce.go

```diff
@@ -435,9 +435,15 @@ func CreateGCECloud(config *CloudConfig) (*Cloud, error) {
 	serviceAlpha.UserAgent = userAgent
 
 	if config.APIEndpoint != "" {
-		service.BasePath = config.APIEndpoint
-		serviceBeta.BasePath = strings.Replace(config.APIEndpoint, "v1", "beta", -1)
-		serviceAlpha.BasePath = strings.Replace(config.APIEndpoint, "v1", "alpha", -1)
+		if strings.HasSuffix(service.BasePath, "/projects/") {
+			service.BasePath = config.APIEndpoint + "projects/"
+			serviceBeta.BasePath = strings.Replace(config.APIEndpoint, "v1", "beta", -1) + "projects/"
+			serviceAlpha.BasePath = strings.Replace(config.APIEndpoint, "v1", "alpha", -1) + "projects/"
```

Comment:
suggest using the helper method that handles trailing slash (or lack of it) correctly:

```go
service.BasePath = getProjectsBasePath(config.APIEndpoint)
serviceBeta.BasePath = getProjectsBasePath(strings.Replace(config.APIEndpoint, "v1", "beta", -1))
serviceAlpha.BasePath = getProjectsBasePath(strings.Replace(config.APIEndpoint, "v1", "alpha", -1))
```

Reply:
Thanks. It makes sense. @liggitt 
Updated.